pkg/volume/azure_dd/common.go

```diff
@@ -0,0 +1,322 @@
+	diskUri = STRINGS.ToLower(diskUri) // always lower uri because users may enter it in caps.
+	uniqueDiskNameTemplate := "%s%s"
+	az, _ := host.GetCloudProvider().(*azure.Cloud)
+	hashedDiskUri := az.MakeCRC32(diskUri)
+	prefix := "b"
+	if isManaged {
+		prefix = "m"
+	}
+	// "{m for managed b for blob}{hashed diskUri or DiskId depending on disk kind }"
+	diskName := fmt.Sprintf(uniqueDiskNameTemplate, prefix, hashedDiskUri)
+	pdPath := path.Join(host.GetPluginDir(azureDataDiskPluginName), mount.MountsInGlobalPDPath, diskName)
+
+	return pdPath, nil
+}
+
+func diskKindHashfromPDName(diskName string) (bool, string) {
+	diskKind := diskName[0:1]
+	diskHash := diskName[1:]
+
+	return (diskKind == "m"), diskHash
+}
+
+func makeDataDisk(volumeName string, podUID types.UID, diskName string, host volume.VolumeHost) *dataDisk {
+	var metricProvider volume.MetricsProvider
+	if podUID != "" {
+		metricProvider = volume.NewMetricsStatFS(getPath(podUID, volumeName, host))
+	}
+
+	return &dataDisk{
+		MetricsProvider: metricProvider,
+		volumeName:      volumeName,
+		diskName:        diskName,
+		podUID:          podUID,
+	}
+}
+
+func getVolumeSource(spec *volume.Spec) (*v1.AzureDiskVolumeSource, error) {
+	if spec.Volume != nil && spec.Volume.AzureDisk != nil {
+		return spec.Volume.AzureDisk, nil
+	}
+
+	if spec.PersistentVolume != nil && spec.PersistentVolume.Spec.AzureDisk != nil {
+		return spec.PersistentVolume.Spec.AzureDisk, nil
+	}
+
+	return nil, fmt.Errorf("azureDisk - Spec does not reference an Azure disk volume type")
+}
+
+func normalizeFsType(fsType string) string {
+	if fsType == "" {
+		return default_fstype
+	}
+
+	return fsType
+}
+
+func normalizeKind(kind v1.AzureDataDiskKind) (v1.AzureDataDiskKind, error) {
+	if kind == "" {
+		return v1.AzureSharedBlobDisk, nil
+	}
+
+	if !supportedDiskKinds.Has(string(kind)) {
+		return "", fmt.Errorf("azureDisk - %s is not supported disk kind. Supported values are %s", kind, supportedDiskKinds.List())
+	}
+
+	return v1.AzureDataDiskKind(kind), nil
+}
+
+func normalizeStorageAccountType(storageAccountType string) (string, error) {
+	if storageAccountType == "" {
+		return default_storage_account_type, nil
+	}
+
+	if !supportedStorageAccountTypes.Has(storageAccountType) {
+		return "", fmt.Errorf("azureDisk - %s is not supported sku/storageaccounttype. Supported values are %s", storageAccountType, supportedStorageAccountTypes.List())
+	}
+
+	return storageAccountType, nil
+}
+
+func normalizeCachingMode(cachingMode v1.AzureDataDiskCachingMode) (v1.AzureDataDiskCachingMode, error) {
+	if cachingMode == "" {
+		return v1.AzureDataDiskCachingReadWrite, nil
+	}
+
+	if !supportedCachingModes.Has(string(cachingMode)) {
+		return "", fmt.Errorf("azureDisk - %s is not supported cachingmode. Supported values are %s", cachingMode, supportedCachingModes.List())
+	}
+
+	return cachingMode, nil
+}
+
+type ioHandler interface {
```

Comment:
move ioHandler and below out of this file - they are going to be consolidated with other volume plugins.